client/views/admin/apps/AppDetailsPage.tsx

```diff
@@ -62,10 +64,10 @@ function AppDetailsPage({ id }) {
 			</Page.Header>
 			<Page.ScrollableContentWithShadow>
 				<Box maxWidth='x600' w='full' alignSelf='center'>
-					{isLoading && <LoadingDetails />}
-					{!isLoading && (
+					{!data && <LoadingDetails />}
+					{data && (
 						<>
-							<AppDetailsPageContent data={data} />
+							<AppDetailsPageContent app={data} />
```

Comment:
I think that lines 67 to 74 can be improved by breaking these validations into small functions to make this easy to be read.